java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -254,6 +255,17 @@ public static native Dataset commitAppend(
    */
   public static native void drop(String path, Map<String, String> storageOptions);
 
+  public void addColumns(
+      SqlExpressions sqlExpressions, Optional<List<String>> readColumns, Optional<Long> batchSize) {
```

Comment:
`readColumns` isn't used when you have `sqlExpresssions` so you can remove it from this overload.